src/main/java/com/pccw/cloud/consumerapp/KafkaCloudListener.java

```diff
@@ -46,5 +49,24 @@ public void readProdffer(String message) throws JsonProcessingException {
         log.info("Product offer message: " + message);
     }
 
+    @KafkaListener(topics = "CustomerUpdate", groupId = "customerUpdate", containerFactory = "customerUpdateFactory")
+    public void customerUpdate(String message) throws JsonProcessingException {
+        MessageDto messageDto = messageService.createMessageDto(message);
 
+        String topic = messageDto.getTopic();
+        switch (topic) {
```

Comment:
Change to switch(messageDto.getTopic()){

Reply:
updated. thanks for feedback